MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -226,30 +354,18 @@ public override async Task<CompletionContext> GetCompletionContextAsync (IAsyncC
 
 			return await base.GetCompletionContextAsync (session, trigger, triggerLocation, applicableToSpan, token);
 		}
-
-		Task<CompletionContext> GetPackageNameCompletions (MSBuildRootDocument doc, int startIdx, int triggerLength)
+		/*
+		async Task<CompletionContext> GetPackageNameCompletions (MSBuildRootDocument doc, MSBuildResolveResult rr, int startIdx, int triggerLength)
 		{
-			/*
-			string name = ((IXmlParserContext)Tracker.Engine).KeywordBuilder.ToString ();
-			if (string.IsNullOrWhiteSpace (name)) {
-				return null;
-			}
+			return await GetPackageNamesAsync (rr, doc);
+		}*/
 
-			return Task.FromResult<ICompletionDataList> (
-				new PackageNameSearchCompletionDataList (name, PackageSearchManager, doc.GetTargetFrameworkNuGetSearchParameter ()) {
-					TriggerWordStart = startIdx,
-					TriggerWordLength = triggerLength
-				}
-			);
-			*/
-			return Task.FromResult (CompletionContext.Empty);
-		}
 
 		async Task<CompletionContext> GetPackageVersionCompletions (MSBuildRootDocument doc, MSBuildResolveResult rr, CancellationToken token)
 		{
 			var packageId = rr.XElement.Attributes.FirstOrDefault (a => a.Name.Name == "Include")?.Value;
 			if (string.IsNullOrEmpty (packageId)) {
-				return null;
+				return CompletionContext.Empty;//return null
```

Comment:
What is the purpose of this change?

Reply:
In functions such as `GetElementCompletionsAsync` and `GetAttributeCompletionsAsync`, all the null checking returns `CompletionContext.Empty;`. I thought returning an empty completion list is more accurate than returning `null`.

Reply:
@mhutch should I change null checking to returning `null` or let it to be `CompletionContext.Empty`?

Reply:
Oh nevermind. Realized that you already requested these changes.

Reply:
I changed it. Let me know if you want me to change null checking else where too?